src/main/java/com/minecolonies/coremod/colony/Structures.java

```diff
@@ -590,6 +595,94 @@ public static void setMD5s(final Map<String, String> md5s)
         }
     }
 
+    /**
+     * Handle a schematic which has been cut into pieces.
+     * This method is valid on the server
+     * The schematic will be gathered until all pieces have been put together and then handled like on the client.
+     *
+     * @param bytes representing the schematic.
+     * @param id UUID.
+     * @param piece the piece.
+     * @param pieces the amount of pieces.
+     */
+    public static boolean handleSaveSchematicMessage(final byte[] bytes, final UUID id, final int pieces, final int piece)
+    {
+        for(final Map.Entry<UUID, Tuple<Long, Map<Integer, byte[]>>> entry: new HashSet<>(schematicPieces.entrySet()))
+        {
+            if(MathUtils.nanoSecondsToSeconds(System.nanoTime() - entry.getValue().getFirst()) > SECONDS_A_MINUTE)
+            {
+                schematicPieces.remove(entry.getKey());
+                Log.getLogger().warn("Waiting too long for piece of structure, discarding it");
+            }
+        }
+        
+        if(pieces == 1)
+        {
+            return Structures.handleSaveSchematicMessage(bytes);
+        }
+        else
+        {
+            if (!canStoreNewSchematic())
+            {
+                Log.getLogger().warn("Could not store schematic in cache");
+                return false;
+            }
+            Log.getLogger().info("Recieved piece: " + piece + " of: " + pieces + " with the size: " + bytes.length + " and ID: " + id.toString());
+            final Map<Integer, byte[]> schemPieces;
+            if(schematicPieces.containsKey(id))
+            {
+                final Tuple<Long, Map<Integer, byte[]>> schemTuple = schematicPieces.remove(id);
+                schemPieces = schemTuple.getSecond();
+
+                if(MathUtils.nanoSecondsToSeconds(System.nanoTime() - schemTuple.getFirst()) > 60)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 60 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)